src/main/java.ru.javarush.cryptoanalyzer/CryptoAnalyzerAlphabet.java

```diff
@@ -0,0 +1,29 @@
+package main.java.ru.javarush.cryptoanalyzer;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+class CryptoAnalyzerAlphabet {
+
+    private CryptoAnalyzerAlphabet() {
+    }
+
+    static final String ALPHABET_SYMBOLS = "abcdefghijklmnpqrstuvwxyzABCDEFGHIJKLRSTUVWXYZ.,:-!? ";
+    static final int ALPHABET_LENGTH = ALPHABET_SYMBOLS.length();
+
+    static List<Character> getAlphabetSymbolsList() {
+        return ALPHABET_SYMBOLS.chars().mapToObj(e -> (char) e).collect(Collectors.toList());
+    }
+
+    static int getPositionByChar(char letter) {
+        return getAlphabetSymbolsList().indexOf(letter);
+    }
+
+    static char getCharByPosition(int position) {
+        int normalizedPosition = position % ALPHABET_LENGTH;
+        if (normalizedPosition < 0) {
+            normalizedPosition += ALPHABET_LENGTH;
+        }
+        return getAlphabetSymbolsList().get(normalizedPosition);
```

Comment:
На каждую позицию мы вызываем довольно сложную функцию только для того, чтобы воспользоваться частичным результатом.
```suggestion
        // где-то раньше
        cachedList =  getAlphabetSymbolsList();
        
        // тут
        return cachedList.get(normalizedPosition);
```

Вычисления функции могут быть сохранены либо в виде поля класса либо создаться в начале процесса кодирования-декодирования и передаваться как параметры к функции